In [1]:
import pandas as pd
import numpy as np
import os.path

In [2]:
data = pd.read_csv('cars.csv',encoding='latin-1')
data.head()

Make       Model   CC  Used since          Km Transmission       City  \
0    Skoda       Rapid  NaN      2014.0   89,000 Km    automatic  Nasr city   
1    Skoda  Octavia A7  NaN      2015.0  120,000 Km    automatic      Cairo   
2  Renault     Sandero  NaN      2016.0   70,000 Km    automatic   El Haram   
3       MG          ZS  NaN      2020.0   21,000 Km    automatic   El Haram   
4       MG           5  NaN      2021.0   28,000 Km    automatic   El Haram   

   Color Body Style Fuel  ...  Alloy wheels  Rear camera    GPS  Fog light  \
0  Black        NaN  gas  ...         False        False  False       True   
1   Blue        NaN  gas  ...          True         True   True       True   
2    Red        NaN  gas  ...         False        False  False      False   
3    Red        NaN  gas  ...          True        False   True       True   
4    Red        NaN  gas  ...          True        False  False       True   

   Rear spoiler  Cruise control  Power Seats  Central lock  Alarm        Price  
0         False           False         True          True   True  225,000 EGP  
1          True            True         True          True   True  295,000 EGP  
2         False           False        False         False  False  200,000 EGP  
3         False            True         True          True   True  360,000 EGP  
4         False           False         True         False   True  340,000 EGP  

[5 rows x 51 columns]

In [3]:
print(f'There are {data.shape[0]} rows and {data.shape[1]} columns')

There are 38611 rows and 51 columns


In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 38611 entries, 0 to 38610
Data columns (total 51 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   Make                        37745 non-null  object 
 1   Model                       37745 non-null  object 
 2   CC                          19067 non-null  object 
 3   Used since                  37745 non-null  float64
 4   Km                          37745 non-null  object 
 5   Transmission                37745 non-null  object 
 6   City                        37745 non-null  object 
 7   Color                       37745 non-null  object 
 8   Body Style                  21285 non-null  object 
 9   Fuel                        37745 non-null  object 
 10  Leather seats               38611 non-null  bool   
 11  Closing mirrors             38611 non-null  bool   
 12  Intelligent parking system  38611 non-null  bool   
 13  Sunroof                     386

there are many null values in our dataset , we have 40 Boolean Column and 10 objects

In [5]:
data.isnull().sum()

Make                            866
Model                           866
CC                            19544
Used since                      866
Km                              866
Transmission                    866
City                            866
Color                           866
Body Style                    17326
Fuel                            866
Leather seats                     0
Closing mirrors                   0
Intelligent parking system        0
Sunroof                           0
Fabric brushes                    0
Air Conditioning                  0
Power Steering                    0
Remote Keyless                    0
Electric mirrors                  0
Front Power Windows               0
Back Power Windows                0
Tinted Glass                      0
CD Player                         0
DVD Player                        0
Bluetooth                         0
Multifunction                     0
Cassette Radio                    0
AUX                         

we have 866 null values in price Columns , let's drop it

In [12]:
data = data[data['Price'].notna()]

In [13]:
# checking 
data.isnull().sum()

Make                              0
Model                             0
CC                            18678
Used since                        0
Km                                0
Transmission                      0
City                              0
Color                             0
Body Style                    16460
Fuel                              0
Leather seats                     0
Closing mirrors                   0
Intelligent parking system        0
Sunroof                           0
Fabric brushes                    0
Air Conditioning                  0
Power Steering                    0
Remote Keyless                    0
Electric mirrors                  0
Front Power Windows               0
Back Power Windows                0
Tinted Glass                      0
CD Player                         0
DVD Player                        0
Bluetooth                         0
Multifunction                     0
Cassette Radio                    0
AUX                         

In [15]:
data['CC'].unique()

array([nan, '1600 CC', '1400 CC', '2000 CC', '1500 CC', '1300 CC',
       '1200 CC', '1000 CC', '1100 CC', '2500 CC', '3000 CC', '4400 CC',
       '4700 CC', '2002 CC', '1800 CC', '900 CC', '4000 CC', '950 CC',
       '3700 CC', '3500 CC', '800 CC', '2300 CC', '2700 CC', '2200 CC',
       '3800 CC', '1989 CC', '3750 CC', '1399 CC', '3400 CC', '4100 CC',
       '4800 CC', '3646 CC', '1250 CC', '2800 CC', '1398 CC', '4500 CC',
       '3200 CC', '5700 CC', '2600 CC', '3600 CC', '2900 CC', '6000 CC',
       '2400 CC', '5200 CC', '1599 CC', '1716 CC', '1116 CC', '1900 CC',
       '2019 CC', '1998 CC', '4200 CC', '1999 CC', '1550 CC', '1323 CC',
       '995 CC', '1598 CC', '1897 CC', '1489 CC', '3300 CC', '1992 CC',
       '5600 CC', '3190 CC', '850 CC', '1139 CC', '2127 CC', '1295 CC',
       '1700 CC', '5400 CC', '1490 CC', '970 CC', '5800 CC', '1050 CC',
       '1850 CC', '1572 CC', '2609 CC', '4570 CC', '1970 CC', '1150 CC',
       '5000 CC', '4600 CC', '1595 CC', '5900 CC', '2100 CC', '